### PR TITLE
[debugging] aot_eager backend to use the min-cut partitioner

### PR DESCRIPTION
**Pull request by anijain2305**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #103632
* __->__ #103555

default_partitioner is kind of broken when it comes to memory footprint. Moving aot_eager to use min-cut partitioner is better debugging experience.

One bad thing though would be that we will much lower speedup numbers, because min cut partitioner will try to recompute ops.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @msaroufim

### CODE REVIEW

**Review by eellison — APPROVED** [review]

**Review by jansel — APPROVED** [review]

**Comment by anijain2305** [comment]
I will wait for @Chillee approval on this. One of the test case was failing because we had a path from source to sink with infinite flow. I broke that path. But not totally sure if thats the right fix.

**Pull request closed without merging**
